src/main/java/org/wso2/carbon/securevault/hashicorp/repository/HashiCorpSecretRepository.java

```diff
@@ -255,15 +307,45 @@ private void retrieveRootToken() throws HashiCorpVaultException {
     private void setTextFileName() {
 
         String osName = System.getProperty("os.name");
-        if (osName.toLowerCase().contains("win")) {
-            textFileName = "hashicorpRootToken.txt";
-            textFileName_tmp = "hashicorpRootToken-tmp.txt";
-            textFilePersist = "hashicorpRootToken-persist.txt";
+
+        if (authType.equals(AuthType.APP_ROLE)) {
+            textFileName = "hashicorpSecretId";
+            textFileName_tmp = "hashicorpSecretId-tmp";
+            textFilePersist = "hashicorpSecretId-persist";
         } else {
             textFileName = "hashicorpRootToken";
             textFileName_tmp = "hashicorpRootToken-tmp";
             textFilePersist = "hashicorpRootToken-persist";
         }
+
+        if (osName.toLowerCase().contains("win")) {
+            textFileName = textFileName + TXT;
+            textFileName_tmp = textFileName_tmp + TXT;
+            textFilePersist = textFilePersist + TXT;
+        }
+    }
+
+    /**
+     * Get the secret id of the vault. Either by prompting the user via the console or by accessing the text file
+     * containing the secret id.
+     */
+    private String retrieveServiceToken(String roleId, String secretId) throws HashiCorpVaultException {
+        try {
+            final VaultConfig config = new VaultConfig().address(address).build();
+
+            Vault vault = new Vault(config);
+            AuthResponse response = vault.auth().loginByAppRole(roleId, secretId);
+
+            // Store the service token
+            accessToken = response.getAuthClientToken();
```

Comment:
Is there any special reason to assign the value to the global variable? You are anyway doing the same when the method is returning the value.

Reply:
Assigned this to local variable and returned it in 913abdc33b35c37b20e73ed5189933bf23648a05